#! /usr/bin/env python
# Task Long configuration

PACKAGE='sift_detect_holo'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("scaleRotRob",      double_t, 0,    "Rotation scale for body",  0.15, 0.0, 1.0)
gen.add("scaleRotCam",      double_t, 0,    "Rotation scale for camera",  0.3, 0.0, 1.0)
gen.add("sigmaVel",           double_t, 0,    "Sigma for linear speed",  0.17, 0.0, 0.6)
gen.add("sigmaRot",           double_t, 0,    "Sigma for linear speed",  0.17, 0.0, 10)
gen.add("minVel",       double_t, 0,    "min velocity before stopping",  0.01, 0.0, 0.05)
gen.add("minRot",  double_t, 0,    "min velocity to have a non zero rotation speed",  0.027, 0.0, 0.15)
gen.add("maxRotRob",  double_t, 0,    "min velocity to have a non zero rotation speed",  0.5, 0.0, 1.0)
gen.add("maxRotCam",  double_t, 0,    "min velocity to have a non zero rotation speed",  0.5, 0.0, 1.0)
gen.add("scale_lin",  double_t, 0,    "global scaling for all speed",  1.5, 0.0, 2)


exit(gen.generate(PACKAGE, "sift_detect_holo", "VelConvert"))

